### PR TITLE
docs(markdown-format): create adr for markdown format

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

there is no ADR/documentation around what flavor of markdown that is supported by https://stenciljs.com/docs/docs-readme
 
GitHub Issue Number: N/A

## What is the new behavior?

create an adr discussing how/why the team shall adopt CommonMark as the
standard for the docs-readme output target

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

I did not - mwahahahahah

### CODE REVIEW

File: docs/adr/0012-output-markdown-format.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      3. Using a well adopted standard may lead to less chance of Markdown renderings not working as intended
+   2. Cons
+      1. Not everyone may be familiar with CommonMark, leading to confusion when reviewing pull requests
+      2. CommonMark has yet to produce a v1.0 specification
+3. Adopt [GitHub Flavored Markdown (GFM)](https://www.iana.org/assignments/markdown-variants/GFM)
+   1. Pros:
+      1. Individuals may be more familiar with GFM, given the popularity of the platform
+      2. Using a well adopted standard may lead to less chance of Markdown renderings not working as intended
+   2. Cons:
+      1. GFM is less defined compared to CommonMark, resulting in potential ambiguities
```
Comment (ltm) [inline docs/adr/0012-output-markdown-format.md:36]: Modern GFM is well-defined and a [strict superset of CommonMark](https://github.blog/2017-03-14-a-formal-spec-for-github-markdown/). However, the [current GFM spec](https://github.github.com/gfm/) is based on version 0.29 of CommonMark while the latest version of [CommonMark](https://spec.commonmark.org/) is 0.30.

File: docs/adr/0012-output-markdown-format.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+## Options
+
+1. [Describe the many options, their pros/cons, side effects, and any perspective a given team member has]
+2. Adopt [CommonMark](https://commonmark.org/)
+   1. Pros:
+      1. It has industry support from the likes of GitHub, GitLab, Qt, and more
+      2. It seems to be the leading effort to create an unambiguous specification for Markdown
+      3. Using a well adopted standard may lead to less chance of Markdown renderings not working as intended
+   2. Cons
```
Comment (ltm) [inline docs/adr/0012-output-markdown-format.md:28]: Another major drawback of CommonMark is that it doesn't define a syntax for tables, so tables have to be produced as [HTML blocks](https://spec.commonmark.org/0.30/#html-blocks).

File: docs/adr/0012-output-markdown-format.md (hunk 1 above)
Comment (rwaskiewicz) [reply docs/adr/0012-output-markdown-format.md]: TIL - I was looking at an old IANA doc. Fixed in 675f1d4075dde8c7c264d07ec42cb8255c3b5ea9

File: docs/adr/0012-output-markdown-format.md (hunk 2 above)
Comment (rwaskiewicz) [reply docs/adr/0012-output-markdown-format.md:28]: Added in b59a91c1a7bcfa79be646ac0a464f455f07db070


To be fair, I dare say I prefer HTML tables in this scenario. I can never remember how to make a table in MD

**Review by ltm — APPROVED** [review]

**Pull request merged**